test/common/http/header_map_impl_test.cc

```diff
@@ -363,7 +363,7 @@ TEST(HeaderStringTest, All) {
 }
 
 TEST(HeaderMapImplTest, InlineInsert) {
-  HeaderMapImpl headers;
+  TestHeaderMapImpl headers;
```

Comment:
IMO this file should use the real header map to the greatest extent possible. It's a small and possibly pedantic nit, but for this file only, can you define an overridden header map impl and move the verify size here? That way we make sure that we aren't accidentally testing test code that we don't expect?

Reply:
This is a good idea; as it stands you might wind up depending on TestHeaderMapImpl for functional correctness with a future change, and have no unit tests that expose that.

Reply:
Yep! Feeling grateful that all I had to do was override the `verifyByteSize()`. 

There is one "testing test code" test though. It tests `TestHeaderMapImpl`'s copy constructor. @htuch